go.mod

```diff
@@ -16,12 +16,12 @@ require (
 	github.com/prometheus/client_golang v1.14.0
 	golang.org/x/crypto v0.1.0
 	golang.org/x/tools v0.3.0
-	k8s.io/api v0.26.0
-	k8s.io/apimachinery v0.26.0
-	k8s.io/cli-runtime v0.26.0
-	k8s.io/client-go v0.26.0
+	k8s.io/api v0.26.1
+	k8s.io/apimachinery v0.26.1
+	k8s.io/cli-runtime v0.26.1
```

Comment:
@justinsb - Would it be possible to remove these dependencies (added via kstatus) 
This pulls in kubectl aswell.
Looking for a trimmed import list.

Reply:
Yes, I think that's the plan. The kstatus methods used here do not depend on `kubectl`.  It's the other methods that introduces the `kubectl`. So remove the `kubectl` dependency may require some kstatus pkg refactoring (not sure how feasible that is). Or we can fork the kstatus in this k-d-p repo.   

I probably should move this PR back to the WIP status, since I'm not planning to work on the applyset yet (I was on some other stuff related to kstatus and `status.condition` and I saw the TODO comment in this file about using kstatus)